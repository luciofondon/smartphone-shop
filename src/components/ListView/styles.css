.list-items {
    width: 100%;
    margin: auto;
    padding: 10px;
    
    display: grid;
    grid-template-columns: repeat(auto-fill, 25em);
    grid-column-gap: 10px;
    grid-row-gap: 10px;


}

.search-items {
    height: 80px;
    float: right;
    width: 100%;
    margin-top: 20px;
    text-align: right;
    padding-right: 60px;
}