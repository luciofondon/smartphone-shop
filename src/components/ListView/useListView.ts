import { useEffect, useState } from "react";
import useFetch from "../../hooks/useFetch";
import { IProduct } from "./components/Item/interfaces";



const useListView = () => {
    //const [products, setProducts] = useState<IProduct[]>([]);
    const [productsFilter, setProductsFilter] = useState<IProduct[]>([]);
    const [textFilter, setTextFilter] = useState<string>("");
    const { data } = useFetch<IProduct[] | []>(`https://front-test-api.herokuapp.com/api/product`, "list", [])

    useEffect(() => {
        // (async () => {
        //     const response = await fetch(`https://front-test-api.herokuapp.com/api/product`);
        //     const responseJson = await response.json();
        //     setProducts(responseJson)
        //     setProductsFilter(responseJson)
        // })();
    }, []);

    useEffect(() => {
        setProductsFilter(data.filter(product => product.brand.toLocaleLowerCase().startsWith(textFilter.toLocaleLowerCase())
            || product.model.toLocaleLowerCase().startsWith(textFilter.toLocaleLowerCase())))
    }, [textFilter, data]);



    const handleChangeFilter = (text: string) => {
        setTextFilter(text)
    }

    return {
        states: {
            products: productsFilter,
        },
        actions: {
            handleChangeFilter
        }
    }
}

export default useListView;