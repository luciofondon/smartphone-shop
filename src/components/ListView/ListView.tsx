import React, { useEffect, useState } from 'react';
import EmptyState from './components/EmptyState';

import Item from './components/Item';
import Search from './components/Search';
import useListView from './useListView';
import './styles.css'


const ListView = () => {

    const {
        states: { products },
        actions: { handleChangeFilter }
    } = useListView();

    return (
        <>
            <div className='search-items'>
                <Search handleChangeFilter={handleChangeFilter} />
            </div>

            <div className='list-items'>
                {
                    products.length === 0 ?
                        <EmptyState /> :
                        products.map(product => <Item product={product} key={product.id} />)
                }

            </div>
        </>
    );
}

export default ListView;
