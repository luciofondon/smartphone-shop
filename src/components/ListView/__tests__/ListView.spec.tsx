import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import ListView from '../ListView';

jest.mock('../useListView', () => () => ({
  states: {
    products: [
      {
        id: '1',
        model: 'x20',
        brand: 'Alcatel',
        imgUrl: 'string',
        price: 200,
      },
    ],
  },
  actions: { handleChangeFilter: jest.fn() },
}));

test('render ListView', () => {
  const { container } = render(
    <Router>
      <ListView />
    </Router>,
  );
  expect(container).toMatchSnapshot();
});
