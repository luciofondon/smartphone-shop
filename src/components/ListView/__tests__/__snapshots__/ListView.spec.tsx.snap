// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ListView 1`] = `
<div>
  <div
    class="search-items"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":r0:"
        id=":r0:-label"
      >
        Buscador
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Buscador
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="list-items"
  >
    <div
      class="item"
    >
      <a
        href="/detail/1"
      >
        <div
          class="img"
        >
           
          <img
            height="110"
            src="string"
            width="80"
          />
        </div>
        <div
          class="detail"
        >
          <div
            class="model"
          >
            x20
          </div>
          <div
            class="brand"
          >
            Alcatel
          </div>
          <div
            class="price"
          >
             
            200
            â‚¬
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
